src/util/generate/detect-from-extensions.ts

```diff
@@ -0,0 +1,120 @@
+import { sep, join } from 'path'
+import { Build } from './generate-project'
+import { DirMap } from './generate-dir-map'
+import { chooseType } from './helpers'
+import { locale, extensions, builders } from './metadata'
+
+export function getCountAndDepth(extension: string, map: DirMap): { depth: number, count: number } {
+  let depth = 0
+  let diveDepth = 0
+  let count = 0
+
+  const extensions = map.extensions[extension]
+  if (extensions) {
+    depth++
+    if (typeof extensions === 'number') {
+      count += extensions
+    } else {
+      count++
+    }
+
+    Object.keys(map.dir).forEach((name) => {
+      const dive = getCountAndDepth(extension, map.dir[name])
+      if (diveDepth < dive.depth) diveDepth = dive.depth
+      count += dive.count
+    })
+    depth += diveDepth
+  }
+
+  return { depth, count }
+}
+
+export function getExtensionOptions(extension: string, type: 'single' | 'many', recovery: 'destructure' | 'manual') {
+  const upload = locale.upload[type]
+  const ignore = locale.ignore[type]
+  const options = extensions[extension]
+    ? extensions[extension].map((builder) => ({
+        [builder]: locale[builder]
+          ? locale[builder][type]
+          : `${type === 'single' ? 'This file' : 'These files'} should be built`
+      })).reduce((acc, val) => Object.assign(acc, val, {}))
+    : {}
+
+  return {
+    ...options,
+    upload,
+    ignore,
+    [recovery]: locale[recovery][type]
+  }
+}
+
+export function getManualOptions(type: 'single' | 'many'): { [key: string]: string } {
+  let options: { [key: string]: string } = {}
+  Object.values(builders).forEach((builder) => {
+    options[builder] = locale[builder]
+      ? locale[builder][type]
+      : `${type === 'single' ? 'This file' : 'These files'} should be built by ${builder}`
+  })
+
+  return options
+}
+
+type helpers = { choose: chooseType }
+export async function detectFromExtensions(map: DirMap, deepCapture: string[], rel: string, { choose }: helpers) {
+  const builds: Build[] = []
+  const capture: string[] = []
+
+  for (let ext in map.extensions) {
+    if (Object.prototype.hasOwnProperty.call(map.extensions, ext) && !deepCapture.includes(ext)) {
+      const { depth, count } = getCountAndDepth(ext, map)
+      let use
+      let src
+
+      if (depth > 1) {
+        use = await choose(
+          `What are the ${count} ${ext} files in .${sep}${rel} (${depth} deep)?`,
```

Comment:
I came upon a really strange question

> What are the 2  files in ./ (shallow)? (Use arrow keys)

I guess I have to files without extensions but I have no idea what they are from this prompt. I checked the repo and its `.gitignore` and `.nowignore`.

Perhaps we should simply skip these two files since we don't really need to prompt the user here. These files are meta.

Reply:
I'll add these files to the `.nowignore` :+1:

Reply:
Changed to `What are the 2 extensionless files in ./ (shallow)? (Use arrow keys)` in https://github.com/zeit/now-cli/pull/2276/commits/ac3496f8ab564470bf5ff4926c567b026150ec0b